### ISSUE TITLE
RPM won't install with python 2.5 

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#305]
Original [issue 305](https://code.google.com/p/pychess/issues/detail?id=305) reported by [dangerousdrdave@hotmail.com](https://code.google.com/u/dangerousdrdave@hotmail.com/) 2008-02-09

**What steps will reproduce the problem?**
1. Download RPM in Fedora 8
2. Double Click on RPM
**3.**

**What is the expected output? What do you see instead?**
I expect the package installer to install pychess. Instead I get a
dependency error requiring Python(abi)=2.4. Fedora 8 is using 2.5.

**Does it happen every time?**
Yes

**What version of the product are you using?**
pychess-0.8beta4.noarch.rpm

Did you use an installed version of PyChess or did you run in from a
tarball/svn?

**Please provide any additional information below.**

**Please attach the latest pychess logfile.**
For PyChess <= 0.6.x it's hidden in your homedirectory and called
".pychess.log".
For PyChess > 0.6.x, it's in a hidden folder, under your homedirectory,
named ".pychess/"

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=305#c1) posted by [dangerousdrdave@hotmail.com](https://code.google.com/u/dangerousdrdave@hotmail.com/) 2008-02-09

I can run pychess manually, but the RPM won't install and setup.py install fails as
well. Both expect Python 2.4, but Fedora 8 uses Python 2.5.

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=305#c2) posted by [lobais](https://code.google.com/u/lobais/) 2008-02-09

Have you tried getting the fedora rpm from yum?

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=305#c3) posted by [dangerousdrdave@hotmail.com](https://code.google.com/u/dangerousdrdave@hotmail.com/) 2008-02-10

Yes, but they are only offering pychess 0.6.0-1.fc7.noarch

If I knew where all the files and folders needed to be placed, I could do it
manually. I might also be curious enough about the whole RPM process to try and build
an RPM myself. Without access to the .spec file in the current RPM I can't even
change python = 2.4 to python >= 2.4. I'll have to research if it's possible to edit
a .spec file in an existing RPM.

**Comment by pychess** [comment]
Original [comment 4.](https://code.google.com/p/pychess/issues/detail?id=305#c4) posted by [lobais](https://code.google.com/u/lobais/) 2008-02-10

0.8 is in fedora 8.
Until then you might consider installing our rpm using --nodeps.

However, I'm not quite sure how the 2.4 dependency occurred in the first place. It
might be because I created the rpm using that version of pychess.
Could you try running: python setup.py bdist_rpm to build an rpm your self. If you
build it with python 2.5 maybe that will be the requirement.

**Comment by pychess** [comment]
Original [comment 5.](https://code.google.com/p/pychess/issues/detail?id=305#c5) posted by [dangerousdrdave@hotmail.com](https://code.google.com/u/dangerousdrdave@hotmail.com/) 2008-02-10

I'm running fedora 8, but all I get in the gnome package manager is pychess
0.6.0-1.fc7.noarch.

Hmmm... I looked in Maximum RPM and found --nodeps (great minds think alike)
ran: # rpm -i -vv --nodeps pychess-0.8beta4.noarch.rpm
It seemed to install: there is now a pychess menu item in games, but clicking on it
gives no response.

I then tried # /usr/bin/pychess but it complained about not being able to import
modules and exited.

tried: # python setup.py bdist_rpm

result: 
...
creating dist
tar -cf dist/pychess-0.8beta4.tar pychess-0.8beta4
gzip -f9 dist/pychess-0.8beta4.tar
removing 'pychess-0.8beta4' (and everything under it)
copying dist/pychess-0.8beta4.tar.gz -> build/bdist.linux-i686/rpm/SOURCES
building RPMs
rpm -ba --define _topdir /home/drdave/pychess-0.8beta4/build/bdist.linux-i686/rpm
--clean build/bdist.linux-i686/rpm/SPECS/pychess.spec
-ba: unknown option
error: command 'rpm' failed with exit status 1 :(

**Comment by pychess** [comment]
Original [comment 6.](https://code.google.com/p/pychess/issues/detail?id=305#c6) posted by [dangerousdrdave@hotmail.com](https://code.google.com/u/dangerousdrdave@hotmail.com/) 2008-02-11

OK...

I installed rpm-build and created /ect/popt with the required lines for rpm.

from # man rpm:

LEGACY ISSUES
   Executing rpmbuild
       The  build  modes  of rpm are now resident in the /usr/bin/rpmbuild
executable. Although legacy compatibility provided by
       the popt aliases below has been adequate, the compatibility is not perfect;
hence build mode compatibility  through  popt
       aliases  is  being  removed from rpm.  Install the rpmbuild package, and see
rpmbuild(8) for documentation of all the rpm
       build modes previously documented here in rpm(8).

```
   Add the following lines to /etc/popt if you wish to continue invoking rpmbuild
```

from the rpm command line:

```
   rpm     exec --bp               rpmb -bp
   rpm     exec --bc               rpmb -bc
   rpm     exec --bi               rpmb -bi
   rpm     exec --bl               rpmb -bl
   rpm     exec --ba               rpmb -ba
```

...

I trashed the old pychess folder and downloaded a fresh copy of the tar.gz file.

I ran: # python setup.py build_rpm

and got:
...
running install
error: invalid Python installation: unable to open /usr/lib/python2.5/config/Makefile
(No such file or directory)
error: Bad exit status from /var/tmp/rpm-tmp.21767 (%install)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.21767 (%install)
error: command 'rpmbuild' failed with exit status 1

It is getting closer, but still complaining about python 2.5.

Man! This is agony! So close and yet so far...

I'm going to let GNU chess kick my butt a few times to restore my humility. :)

**Comment by pychess** [comment]
Original [comment 7.](https://code.google.com/p/pychess/issues/detail?id=305#c7) posted by [dangerousdrdave@hotmail.com](https://code.google.com/u/dangerousdrdave@hotmail.com/) 2008-02-11

OK...

BIG HINT"

error: invalid Python installation: unable to open /usr/lib/python2.5/config/Makefile
(No such file or directory)

So on a hunch I went back to the package manager and looked around for python
development stuff to install. I didn't pay much attention to what I tried but these
are among what I added:

rpm-pythom
python-devel
python-setuptools-devel
python-tools

tried: # python setup.py build_rpm

and it worked!

I got a tar.gz a source rpm and a regular rpm in /dist.

I uninstalled pychess 0.8.0 and then double clicked on the regular rpm.

The install went as planned and now clicking on the pychess menu item in games brings
up python!

I tried downloading and installing the rpm from gnomefiles.org but the package
manager complained about requiring python = 2.4 again.

I'm e-mailing you the new dist folder to your gmail account.

First I'm going to change distro info from fedora 6 to fedora 8 and rebuild the rpm.
It's just a little detail but it matters.
